src/main/java/com/yelp/nrtsearch/server/search/sort/SortParser.java

```diff
@@ -196,7 +196,9 @@ private static SearchResponse.Hit.CompositeFieldValue getValueForSortField(
         break;
       case STRING:
       case STRING_VAL:
-        if (sortValue instanceof BytesRef) {
+        if (sortValue == null) {
+          fieldValue.setTextValue("");
+        } else if (sortValue instanceof BytesRef) {
```

Comment:
Can this condition happen for more than just string fields? How can you distinguish between a unset field and one set with an empty string? Should this return an empty `CompositeFieldValue` instead?

Reply:
I'll try return empty CompositeFieldValue and add tests to see if it works